package cujak.fft2d.pb;

/** double */
message dComplex {
  required double real = 1;
  required double imag = 2;
}

message Property {
  optional double Lx = 1;
  optional double Ly = 2;
  optional uint64 index = 3;
  optional double time = 4;
}

message dCoefficient {
  required uint64 Nx = 1;
  required uint64 Ny = 2;
  repeated dComplex value = 3;
  optional Property property = 4;
} // extention=.dc

message dField {
  required uint64 Nx = 1;
  required uint64 Ny = 2;
  repeated double value = 3;
  optional Property property = 4;
} // extention=.df

/** float */
message fComplex {
  required float real = 1;
  required float imag = 2;
}

message fCoefficient {
  required uint64 Nx = 1;
  required uint64 Ny = 2;
  repeated fComplex value = 3;
  optional Property property = 4;
} // extention=.fc

message fField {
  required uint64 Nx = 1;
  required uint64 Ny = 2;
  repeated float value = 3;
  optional Property property = 4;
} // extention=.ff
